cli/azd/pkg/auth/manager.go

```diff
@@ -111,8 +117,17 @@ func EnsureLoggedInCredential(ctx context.Context, credential azcore.TokenCreden
 }
 
 // CredentialForCurrentUser returns a TokenCredential instance for the current user. If `auth.useLegacyAzCliAuth` is set to
-// a truthy value in config, an instance of azidentity.AzureCLICredential is returned instead.
-func (m *Manager) CredentialForCurrentUser(ctx context.Context) (azcore.TokenCredential, error) {
+// a truthy value in config, an instance of azidentity.AzureCLICredential is returned instead. To accept the default options,
+// pass nil.
+func (m *Manager) CredentialForCurrentUser(
```

Comment:
This func is getting pretty large and you may want to consider breaking it down into smaller funcs for each of the big if conditions.

Reply:
Good call - I will break this apart.